sources/common/utils.go

```diff
@@ -38,11 +40,135 @@ func ToNotNull(conv *internal.Conv, isNullable string) bool {
 func GetColsAndSchemas(conv *internal.Conv, srcTable string) (schema.Table, string, []string, ddl.CreateTable, error) {
 	srcSchema := conv.SrcSchema[srcTable]
 	spTable, err1 := internal.GetSpannerTable(conv, srcTable)
-	spCols, err2 := internal.GetSpannerCols(conv, srcTable, srcSchema.ColNames)
+	spCols, err2 := internal.GetSpannerCols(conv, srcTable, srcSchema.ColIds)
 	spSchema, ok := conv.SpSchema[spTable]
 	var err error
 	if err1 != nil || err2 != nil || !ok {
 		err = fmt.Errorf(fmt.Sprintf("err1=%s, err2=%s, ok=%t", err1, err2, ok))
 	}
 	return srcSchema, spTable, spCols, spSchema, err
 }
+
+func GetSpTableIdFromName(conv *internal.Conv, tableName string) string {
+	for tableId, table := range conv.SpSchema {
+		if tableName == table.Name {
+			return tableId
+		}
+	}
+	return ""
+}
+
+func GetSpColIdFromName(conv *internal.Conv, tableId, colName string) string {
+	for colId, col := range conv.SpSchema[tableId].ColDefs {
+		if col.Name == colName {
+			return colId
+		}
+	}
+	return ""
+}
+
+func getSpFkIdFromName(conv *internal.Conv, tableId, fkName string) string {
+	for _, fk := range conv.SpSchema[tableId].ForeignKeys {
+		if fk.Name == fkName {
+			return fk.Id
+		}
+	}
+	return ""
+}
+
+func getSpIndexFromName(conv *internal.Conv, tableId, indexName string) (ddl.CreateIndex, error) {
+	for _, index := range conv.SpSchema[tableId].Indexes {
+		if index.Name == indexName {
+			return index, nil
+		}
+	}
+	return ddl.CreateIndex{}, fmt.Errorf("spanner index not found")
+}
+
```

Comment:
Only seem to be used in the `assert` methods below, I think they need to be moved out as well.

Reply:
Yes. Private fucntions are moved to internal/test_utils.go and the public ones are moved to internal/helpers.go.